### ISSUE TITLE
RC source-build build fails building coreclr with cmake error on RHEL 7.4

### ISSUE THREAD

**Issue by dleeapho** [dotnet/source-build#512]
`Scanning dependencies of target clrjit
[ 74%] Building CXX object src/utilcode/dyncrt/CMakeFiles/utilcode.dir/__/__/__/version.cpp.o
[ 74%] Linking CXX static library libutilcode.a
[ 74%] Building CXX object src/jit/standalone/CMakeFiles/clrjit.dir/__/alloc.cpp.o
error: Could not read profile: Unsupported format version
[ 74%] Built target utilcode
1 error generated.
make[2]: *** [src/jit/standalone/CMakeFiles/clrjit.dir/__/alloc.cpp.o] Error 1
make[1]: *** [src/jit/standalone/CMakeFiles/clrjit.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
Scanning dependencies of target protononjit`

**Comment by crummel** [comment]
I think this might be a clang error rather than a CMake error.  I have a working machine with CMake 2.8.12.2 and clang 3.9.1 and a repro machine with CMake 3.3.2 and clang 3.6.2.

**Comment by crummel** [comment]
If it's clang, I think it could be fixed by adding the `-nopgooptimize` flag in CoreCLR, trying this now.

**Comment by crummel** [comment]
cc @omajid

**Comment by dleeapho** [comment]
I built using the prereqs Docker image from CI and it built without issues: cmake 3.3.2 and clang 3.6.2 These are the same as my repro machine that uses an older prereqs image.

**Comment by crummel** [comment]
Wild guess based on [this](https://github.com/dotnet/coreclr/blob/7916aa9d417eea94d7cc3a47902f6c626fd72c00/pgosupport.cmake#L30):  Do you have any packages matching `optimization.*.PGO.CoreCLR`?  I'm wondering if they don't get included in the tarball.

**Comment by dleeapho** [comment]
On `prereqs_2` there was no occurrence of `profile-instr-use`. Ergo, PGO wasn't used.

**Comment by crummel** [comment]
Ok, I logged https://github.com/dotnet/source-build/issues/514 to get our environment right and turn PGO back on by default.

**Comment by crummel** [comment]
Fixed by #513.

**Comment by crummel** [comment]
One last follow-up: according to [LLVM](https://llvm.org/docs/GoldPlugin.html#lto-how-to-build), PGO requires LTO which requires LLVMgold which requires `ld` plugin support.  On our CI docker image I get:
```
[root@77810b22880b /]# ld -plugin
ld: unrecognized option '-plugin'
```
which indicates no plugin support, vs my working PGO machine with
```
[chris@crummel-rh2 ~]$ ld -plugin
ld: -plugin: missing argument
```

So that seems to be the root cause of PGO not running on the CI image at all.
(I'll put this in the other issue too).
